include/benchmark/benchmark_api.h

```diff
@@ -609,11 +756,15 @@ class Fixture: public internal::Benchmark {
 
     virtual void Run(State& st) {
       this->SetUp(st);
+      this->InitState(st);
       this->BenchmarkCase(st);
+      this->TerminateState(st);
       this->TearDown(st);
     }
 
     virtual void SetUp(const State&) {}
+    virtual void InitState(State &) {}
```

Comment:
i'm not a fan of the two-stage initialization and shutdown just to support counters in fixtures, no. I suggest that for now we don't support fixtures and get the basic counters in. then we can figure out how to support fixtures better.

it may be ok to pass a mutable State, but i need to think about it.